// @metamask/controller-utils/dist/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworksTicker = exports.NetworkId = exports.NetworksChainId = exports.isNetworkType = exports.NetworkType = void 0;
/**
 * Human-readable network name
 */
var NetworkType;
(function (NetworkType) {
    NetworkType["localhost"] = "localhost";
    NetworkType["mainnet"] = "mainnet";
    NetworkType["goerli"] = "goerli";
    NetworkType["sepolia"] = "sepolia";
    NetworkType["rpc"] = "rpc";
})(NetworkType = exports.NetworkType || (exports.NetworkType = {}));
/**
 * A helper to determine whether a given input is NetworkType.
 *
 * @param val - the value to check whether it is NetworkType or not.
 * @returns boolean indicating whether or not the argument is NetworkType.
 */
function isNetworkType(val) {
    return Object.values(NetworkType).includes(val);
}
exports.isNetworkType = isNetworkType;
var NetworksChainId;
(function (NetworksChainId) {
    NetworksChainId["mainnet"] = "1";
    NetworksChainId["goerli"] = "5";
    NetworksChainId["sepolia"] = "11155111";
    NetworksChainId["localhost"] = "";
    NetworksChainId["rpc"] = "";
})(NetworksChainId = exports.NetworksChainId || (exports.NetworksChainId = {}));
var NetworkId;
(function (NetworkId) {
    NetworkId["mainnet"] = "1";
    NetworkId["goerli"] = "5";
    NetworkId["sepolia"] = "11155111";
})(NetworkId = exports.NetworkId || (exports.NetworkId = {}));
var NetworksTicker;
(function (NetworksTicker) {
    NetworksTicker["mainnet"] = "ETH";
    NetworksTicker["goerli"] = "GoerliETH";
    NetworksTicker["sepolia"] = "SepoliaETH";
    NetworksTicker["localhost"] = "";
    NetworksTicker["rpc"] = "";
})(NetworksTicker = exports.NetworksTicker || (exports.NetworksTicker = {}));
